### ISSUE TITLE
Crashes on Adding navidrome server

### ISSUE THREAD

**Issue by tarasis** [SubmarinerApp/Submariner#141]
**Describe the bug**

I added my navidrome server, and the program immediately crashed. When I reopen it and click on the server in the sidebar the app crashes.

**To Reproduce**

Open Submariner
Click New Server in Menu
Enter server details
Tap OK.

(make no difference if Token Authentication is checked or not)

**Expected behavior**

The app shouldn't crash

**Versions:**
 - macOS: 13.4
 - Subsonic Server and Version: Navidrome 0.49.3
 - Submariner Version: 2.3
 - Download Source: Git / Website

**Additional context**

Able to connect fine with phone apps to the server.

<details>
<summary>Crash output</summary>
```
-------------------------------------
Translated Report (Full Report Below)
-------------------------------------

Process:               Submariner [30389]
Path:                  /Applications/Submariner.app/Contents/MacOS/Submariner
Identifier:            fr.read-write.Submariner
Version:               2.3 (2.3)
Code Type:             ARM-64 (Native)
Parent Process:        launchd [1]
User ID:               501

Date/Time:             2023-06-24 16:58:37.2170 +0100
OS Version:            macOS 13.4 (22F66)
Report Version:        12
Anonymous UUID:        2C3645DF-3452-BDEA-3844-2B62CB80AF0C


Time Awake Since Boot: 16000 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BREAKPOINT (SIGTRAP)
Exception Codes:       0x0000000000000001, 0x0000000102d56344

Termination Reason:    Namespace SIGNAL, Code 5 Trace/BPT trap: 5
Terminating Process:   exc handler [30389]

Thread 0 Crashed::  Dispatch queue: com.apple.main-thread
0   Submariner                    	       0x102d56344 0x102d28000 + 189252
1   Submariner                    	       0x102d5724c 0x102d28000 + 193100
2   Submariner                    	       0x102d7f810 0x102d28000 + 358416
3   Submariner                    	       0x102d341cc 0x102d28000 + 49612
4   Submariner                    	       0x102d362e8 0x102d28000 + 58088
5   CoreFoundation                	       0x19ebdf180 __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 148
6   CoreFoundation                	       0x19ec7aeec ___CFXRegistrationPost_block_invoke + 88
7   CoreFoundation                	       0x19ec7ae34 _CFXRegistrationPost + 440
8   CoreFoundation                	       0x19ebb04cc _CFXNotificationPost + 704
9   Foundation                    	       0x19fb0e8f4 -[NSNotificationCenter postNotificationName:object:userInfo:] + 88
10  Submariner                    	       0x102d3a6b4 0x102d28000 + 75444
11  CoreFoundation                	       0x19ebdf180 __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 148
12  CoreFoundation                	       0x19ec7aeec ___CFXRegistrationPost_block_invoke + 88
13  CoreFoundation                	       0x19ec7ae34 _CFXRegistrationPost + 440
14  CoreFoundation                	       0x19ebb04cc _CFXNotificationPost + 704
15  Foundation                    	       0x19fb0e8f4 -[NSNotificationCenter postNotificationName:object:userInfo:] + 88
16  AppKit                        	       0x1a1efb120 -[NSTableView _sendSelectionChangedNotificationForRows:columns:] + 196
17  AppKit                        	       0x1a20b70c8 -[NSTableView mouseDown:] + 3968
18  AppKit                        	       0x1a20b5fc8 -[NSOutlineView mouseDown:] + 76
19  AppKit                        	       0x1a1fac2cc -[NSWindow(NSEventRouting) _handleMouseDownEvent:isDelayedEvent:] + 3476
20  AppKit                        	       0x1a1f36f08 -[NSWindow(NSEventRouting) _reallySendEvent:isDelayedEvent:] + 364
21  AppKit                        	       0x1a1f36bc8 -[NSWindow(NSEventRouting) sendEvent:] + 284
22  AppKit                        	       0x1a1f35f0c -[NSApplication(NSEvent) sendEvent:] + 1556
23  Submariner                    	       0x102d5c028 0x102d28000 + 213032
24  Submariner                    	       0x102d5c28c 0x102d28000 + 213644
25  AppKit                        	       0x1a2185fc4 -[NSApplication _handleEvent:] + 60
26  AppKit                        	       0x1a1dfd368 -[NSApplication run] + 500
27  AppKit                        	       0x1a1dd4794 NSApplicationMain + 880
28  dyld                          	       0x19e7b3f28 start + 2236

Thread 1:
0   libsystem_pthread.dylib       	       0x19eb06d8c start_wqthread + 0

Thread 2:
0   libsystem_pthread.dylib       	       0x19eb06d8c start_wqthread + 0

Thread 3:
0   libsystem_pthread.dylib       	       0x19eb06d8c start_wqthread + 0

Thread 4:
0   libsystem_pthread.dylib       	       0x19eb06d8c start_wqthread + 0

Thread 5:
0   libsystem_kernel.dylib        	       0x19eacf710 __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x19eb0c574 _pthread_cond_wait + 1232
2   Foundation                    	       0x1a00dd3e4 -[_NSThreadPerformInfo wait] + 44
3   Foundation                    	       0x19fb51db0 -[NSObject(NSThreadPerformAdditions) performSelector:onThread:withObject:waitUntilDone:modes:] + 728
4   CoreFoundation                	       0x19ebcae30 ___forwarding___ + 976
5   CoreFoundation                	       0x19ebca9a0 _CF_forwarding_prep_0 + 96
6   ImageKit                      	       0x1c198af60 -[IKTaskManager taskLoop] + 1316
7   Foundation                    	       0x19fb5b54c __NSThread__start__ + 716
8   libsystem_pthread.dylib       	       0x19eb0bfa8 _pthread_start + 148
9   libsystem_pthread.dylib       	       0x19eb06da0 thread_start + 8

Thread 6:: com.apple.NSEventThread
0   libsystem_kernel.dylib        	       0x19eacbf14 mach_msg2_trap + 8
1   libsystem_kernel.dylib        	       0x19eade240 mach_msg2_internal + 80
2   libsystem_kernel.dylib        	       0x19ead4b78 mach_msg_overwrite + 604
3   libsystem_kernel.dylib        	       0x19eacc290 mach_msg + 24
4   CoreFoundation                	       0x19ebea7e4 __CFRunLoopServiceMachPort + 160
5   CoreFoundation                	       0x19ebe90c4 __CFRunLoopRun + 1208
6   CoreFoundation                	       0x19ebe84b8 CFRunLoopRunSpecific + 612
7   AppKit                        	       0x1a1f34334 _NSEventThread + 172
8   libsystem_pthread.dylib       	       0x19eb0bfa8 _pthread_start + 148
9   libsystem_pthread.dylib       	       0x19eb06da0 thread_start + 8


Thread 0 crashed with ARM Thread State (64-bit):
    x0: 0x0000000000000000   x1: 0x0000000000000000   x2: 0xac081ed59be356b5   x3: 0x0000000000000008
    x4: 0x0000000000100000   x5: 0x000000000000002d   x6: 0x0000000000000000   x7: 0x0000000000000000
    x8: 0x0000000000000000   x9: 0x00000001ee5ef4f4  x10: 0x000000018e4e7f24  x11: 0x000000000000001f
   x12: 0x0000000000000014  x13: 0x0000000142e247a0  x14: 0x0100000122f2cda3  x15: 0x0000000122f2cda0
   x16: 0x000000019e7705e8  x17: 0x0f6e0001a4838f0c  x18: 0x0000000000000000  x19: 0x000000016d0d5790
   x20: 0x0000600001f1c180  x21: 0x3a73697361726174  x22: 0x000000016d0d57a0  x23: 0xe800000000000000
   x24: 0x00000001fabdf6e8  x25: 0x0000000000000010  x26: 0xe700000000000000  x27: 0x00000001fabdfbb8
   x28: 0x0000600003b07c20   fp: 0x000000016d0d5920   lr: 0x0000000102d55f84
    sp: 0x000000016d0d5780   pc: 0x0000000102d56344 cpsr: 0x80001000
   far: 0x0000000239423190  esr: 0xf2000001 (Breakpoint) brk 1

Binary Images:
       0x102d28000 -        0x102db3fff fr.read-write.Submariner (2.3) <aac5136e-2ecf-364e-a323-6b623b284a36> /Applications/Submariner.app/Contents/MacOS/Submariner
       0x10fb3c000 -        0x10fb47fff libobjc-trampolines.dylib (*) <570c31f7-94c6-3b86-ae44-1694db0a4bcd> /usr/lib/libobjc-trampolines.dylib
       0x112234000 -        0x11229ffff com.apple.AppleMetalOpenGLRenderer (1.0) <01596a35-894c-3104-8314-40d3c08f18ca> /System/Library/Extensions/AppleMetalOpenGLRenderer.bundle/Contents/MacOS/AppleMetalOpenGLRenderer
       0x19eb6b000 -        0x19f044fff com.apple.CoreFoundation (6.9) <203e4401-8c2e-3157-a24b-92f52551d43e> /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
       0x19fb05000 -        0x1a04cafff com.apple.Foundation (6.9) <d5337cc6-a075-3ca0-b5bf-b49da8878337> /System/Library/Frameworks/Foundation.framework/Versions/C/Foundation
       0x1a1dd1000 -        0x1a2cdffff com.apple.AppKit (6.9) <041c7d21-21cc-3eea-ae53-d39f5ca6a24e> /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit
       0x19e7ae000 -        0x19e83c553 dyld (*) <bd993f39-c902-37d6-8bd8-6d827c7f121e> /usr/lib/dyld
       0x19eb05000 -        0x19eb11fff libsystem_pthread.dylib (*) <46d35233-a051-3f4f-bba4-ba56dddc4d1a> /usr/lib/system/libsystem_pthread.dylib
       0x19eacb000 -        0x19eb04fe7 libsystem_kernel.dylib (*) <42f503e2-9273-360a-a086-c1b19bbd3962> /usr/lib/system/libsystem_kernel.dylib
       0x1c1965000 -        0x1c1ba8fff com.apple.imageKit (3.0) <b8b17538-9d8d-304f-9fd0-ef28ab78c686> /System/Library/Frameworks/Quartz.framework/Versions/A/Frameworks/ImageKit.framework/Versions/A/ImageKit

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0

VM Region Summary:
ReadOnly portion of Libraries: Total=1.4G resident=0K(0%) swapped_out_or_unallocated=1.4G(100%)
Writable regions: Total=1.5G written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=1.5G(100%)

                                VIRTUAL   REGION 
REGION TYPE                        SIZE    COUNT (non-coalesced) 
===========                     =======  ======= 
Accelerate framework               256K        2 
Activity Tracing                   256K        1 
CG backing stores                 2176K        4 
CG image                          8080K       21 
ColorSync                          496K       25 
CoreAnimation                     24.5M       75 
CoreData                            48K        1 
CoreData Object IDs               4112K        2 
CoreGraphics                        48K        2 
CoreUI image data                 2832K       23 
Foundation                          16K        1 
Image IO                          1024K        1 
Kernel Alloc Once                   32K        1 
MALLOC                           266.1M       56 
MALLOC guard page                  288K       15 
MALLOC_MEDIUM (reserved)         840.0M        7         reserved VM address space (unallocated)
MALLOC_NANO (reserved)           384.0M        1         reserved VM address space (unallocated)
SQLite page cache                  128K        1 
STACK GUARD                       56.1M        7 
Stack                             11.2M        7 
VM_ALLOCATE                       1232K       14 
__AUTH                            1927K      358 
__AUTH_CONST                      29.2M      582 
__CTF                               824        1 
__DATA                            13.0M      568 
__DATA_CONST                      28.4M      588 
__DATA_DIRTY                      2066K      226 
__FONT_DATA                        2352        1 
__GLSLBUILTINS                    5174K        1 
__INFO_FILTER                         8        1 
__LINKEDIT                       802.2M        4 
__OBJC_RO                         66.3M        1 
__OBJC_RW                         2011K        1 
__TEXT                           631.5M      609 
dyld private memory                272K        2 
mapped file                      218.8M       39 
shared memory                      976K       20 
===========                     =======  ======= 
TOTAL                              3.3G     3269 
TOTAL, minus reserved VM space     2.1G     3269 



-----------
Full Report
-----------

{"app_name":"Submariner","timestamp":"2023-06-24 16:58:37.00 +0100","app_version":"2.3","slice_uuid":"aac5136e-2ecf-364e-a323-6b623b284a36","build_version":"2.3","platform":1,"bundleID":"fr.read-write.Submariner","share_with_app_devs":1,"is_first_party":0,"bug_type":"309","os_version":"macOS 13.4 (22F66)","roots_installed":0,"name":"Submariner","incident_id":"CCDD19E3-5211-4BBE-B3CD-BCE9F7C0A19B"}
{
  "uptime" : 16000,
  "procRole" : "Foreground",
  "version" : 2,
  "userID" : 501,
  "deployVersion" : 210,
  "modelCode" : "MacBookAir10,1",
  "coalitionID" : 4750,
  "osVersion" : {
    "train" : "macOS 13.4",
    "build" : "22F66",
    "releaseType" : "User"
  },
  "captureTime" : "2023-06-24 16:58:37.2170 +0100",
  "incident" : "CCDD19E3-5211-4BBE-B3CD-BCE9F7C0A19B",
  "pid" : 30389,
  "translated" : false,
  "cpuType" : "ARM-64",
  "roots_installed" : 0,
  "bug_type" : "309",
  "procLaunch" : "2023-06-24 16:58:34.1806 +0100",
  "procStartAbsTime" : 396431926083,
  "procExitAbsTime" : 396504427102,
  "procName" : "Submariner",
  "procPath" : "\/Applications\/Submariner.app\/Contents\/MacOS\/Submariner",
  "bundleInfo" : {"CFBundleShortVersionString":"2.3","CFBundleVersion":"2.3","CFBundleIdentifier":"fr.read-write.Submariner"},
  "storeInfo" : {"deviceIdentifierForVendor":"F87E4738-0E2F-54EB-B992-2E80EBCDA940","thirdParty":true},
  "parentProc" : "launchd",
  "parentPid" : 1,
  "coalitionName" : "fr.read-write.Submariner",
  "crashReporterKey" : "2C3645DF-3452-BDEA-3844-2B62CB80AF0C",
  "codeSigningID" : "fr.read-write.Submariner",
  "codeSigningTeamID" : "MZ6WW388VL",
  "codeSigningFlags" : 570503953,
  "codeSigningValidationCategory" : 6,
  "codeSigningTrustLevel" : 0,
  "sip" : "enabled",
  "exception" : {"codes":"0x0000000000000001, 0x0000000102d56344","rawCodes":[1,4342506308],"type":"EXC_BREAKPOINT","signal":"SIGTRAP"},
  "termination" : {"flags":0,"code":5,"namespace":"SIGNAL","indicator":"Trace\/BPT trap: 5","byProc":"exc handler","byPid":30389},
  "os_fault" : {"process":"Submariner"},
  "extMods" : {"caller":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"system":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"targeted":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"warnings":0},
  "faultingThread" : 0,
  "threads" : [{"triggered":true,"id":391540,"threadState":{"x":[{"value":0},{"value":0},{"value":12396191877129524917},{"value":8},{"value":1048576},{"value":45},{"value":0},{"value":0},{"value":0},{"value":8294167796,"objc-selector":"didAccessValueForKey:"},{"value":6682476324},{"value":31},{"value":20},{"value":5417093024},{"value":72057598919232931},{"value":4881304992},{"value":6953567720,"symbolLocation":0,"symbol":"objc_release"},{"value":1111826165062143756,"symbolLocation":1111826158007091200,"symbol":"-[NSManagedObject didAccessValueForKey:]"},{"value":0},{"value":6124558224},{"value":105553148887424},{"value":4211826020798587252},{"value":6124558240},{"value":16717361816799281152},{"value":8501720808,"symbolLocation":0,"symbol":"value witness table for Builtin.Int64"},{"value":16},{"value":16645304222761353216},{"value":8501722040,"symbolLocation":0,"symbol":"value witness table for Builtin.UnknownObject"},{"value":105553178164256}],"flavor":"ARM_THREAD_STATE64","lr":{"value":4342505348},"cpsr":{"value":2147487744},"fp":{"value":6124558624},"sp":{"value":6124558208},"esr":{"value":4060086273,"description":"(Breakpoint) brk 1"},"pc":{"value":4342506308,"matchesCrashFrame":1},"far":{"value":9550573968}},"queue":"com.apple.main-thread","frames":[{"imageOffset":189252,"imageIndex":0},{"imageOffset":193100,"imageIndex":0},{"imageOffset":358416,"imageIndex":0},{"imageOffset":49612,"imageIndex":0},{"imageOffset":58088,"imageIndex":0},{"imageOffset":475520,"symbol":"__CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__","symbolLocation":148,"imageIndex":3},{"imageOffset":1113836,"symbol":"___CFXRegistrationPost_block_invoke","symbolLocation":88,"imageIndex":3},{"imageOffset":1113652,"symbol":"_CFXRegistrationPost","symbolLocation":440,"imageIndex":3},{"imageOffset":283852,"symbol":"_CFXNotificationPost","symbolLocation":704,"imageIndex":3},{"imageOffset":39156,"symbol":"-[NSNotificationCenter postNotificationName:object:userInfo:]","symbolLocation":88,"imageIndex":4},{"imageOffset":75444,"imageIndex":0},{"imageOffset":475520,"symbol":"__CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__","symbolLocation":148,"imageIndex":3},{"imageOffset":1113836,"symbol":"___CFXRegistrationPost_block_invoke","symbolLocation":88,"imageIndex":3},{"imageOffset":1113652,"symbol":"_CFXRegistrationPost","symbolLocation":440,"imageIndex":3},{"imageOffset":283852,"symbol":"_CFXNotificationPost","symbolLocation":704,"imageIndex":3},{"imageOffset":39156,"symbol":"-[NSNotificationCenter postNotificationName:object:userInfo:]","symbolLocation":88,"imageIndex":4},{"imageOffset":1220896,"symbol":"-[NSTableView _sendSelectionChangedNotificationForRows:columns:]","symbolLocation":196,"imageIndex":5},{"imageOffset":3039432,"symbol":"-[NSTableView mouseDown:]","symbolLocation":3968,"imageIndex":5},{"imageOffset":3035080,"symbol":"-[NSOutlineView mouseDown:]","symbolLocation":76,"imageIndex":5},{"imageOffset":1946316,"symbol":"-[NSWindow(NSEventRouting) _handleMouseDownEvent:isDelayedEvent:]","symbolLocation":3476,"imageIndex":5},{"imageOffset":1466120,"symbol":"-[NSWindow(NSEventRouting) _reallySendEvent:isDelayedEvent:]","symbolLocation":364,"imageIndex":5},{"imageOffset":1465288,"symbol":"-[NSWindow(NSEventRouting) sendEvent:]","symbolLocation":284,"imageIndex":5},{"imageOffset":1462028,"symbol":"-[NSApplication(NSEvent) sendEvent:]","symbolLocation":1556,"imageIndex":5},{"imageOffset":213032,"imageIndex":0},{"imageOffset":213644,"imageIndex":0},{"imageOffset":3887044,"symbol":"-[NSApplication _handleEvent:]","symbolLocation":60,"imageIndex":5},{"imageOffset":181096,"symbol":"-[NSApplication run]","symbolLocation":500,"imageIndex":5},{"imageOffset":14228,"symbol":"NSApplicationMain","symbolLocation":880,"imageIndex":5},{"imageOffset":24360,"symbol":"start","symbolLocation":2236,"imageIndex":6}]},{"id":391563,"frames":[{"imageOffset":7564,"symbol":"start_wqthread","symbolLocation":0,"imageIndex":7}]},{"id":391564,"frames":[{"imageOffset":7564,"symbol":"start_wqthread","symbolLocation":0,"imageIndex":7}]},{"id":391565,"frames":[{"imageOffset":7564,"symbol":"start_wqthread","symbolLocation":0,"imageIndex":7}]},{"id":391584,"frames":[{"imageOffset":7564,"symbol":"start_wqthread","symbolLocation":0,"imageIndex":7}]},{"id":391643,"frames":[{"imageOffset":18192,"symbol":"__psynch_cvwait","symbolLocation":8,"imageIndex":8},{"imageOffset":30068,"symbol":"_pthread_cond_wait","symbolLocation":1232,"imageIndex":7},{"imageOffset":6128612,"symbol":"-[_NSThreadPerformInfo wait]","symbolLocation":44,"imageIndex":4},{"imageOffset":314800,"symbol":"-[NSObject(NSThreadPerformAdditions) performSelector:onThread:withObject:waitUntilDone:modes:]","symbolLocation":728,"imageIndex":4},{"imageOffset":392752,"symbol":"___forwarding___","symbolLocation":976,"imageIndex":3},{"imageOffset":391584,"symbol":"_CF_forwarding_prep_0","symbolLocation":96,"imageIndex":3},{"imageOffset":155488,"symbol":"-[IKTaskManager taskLoop]","symbolLocation":1316,"imageIndex":9},{"imageOffset":353612,"symbol":"__NSThread__start__","symbolLocation":716,"imageIndex":4},{"imageOffset":28584,"symbol":"_pthread_start","symbolLocation":148,"imageIndex":7},{"imageOffset":7584,"symbol":"thread_start","symbolLocation":8,"imageIndex":7}]},{"id":391645,"name":"com.apple.NSEventThread","frames":[{"imageOffset":3860,"symbol":"mach_msg2_trap","symbolLocation":8,"imageIndex":8},{"imageOffset":78400,"symbol":"mach_msg2_internal","symbolLocation":80,"imageIndex":8},{"imageOffset":39800,"symbol":"mach_msg_overwrite","symbolLocation":604,"imageIndex":8},{"imageOffset":4752,"symbol":"mach_msg","symbolLocation":24,"imageIndex":8},{"imageOffset":522212,"symbol":"__CFRunLoopServiceMachPort","symbolLocation":160,"imageIndex":3},{"imageOffset":516292,"symbol":"__CFRunLoopRun","symbolLocation":1208,"imageIndex":3},{"imageOffset":513208,"symbol":"CFRunLoopRunSpecific","symbolLocation":612,"imageIndex":3},{"imageOffset":1454900,"symbol":"_NSEventThread","symbolLocation":172,"imageIndex":5},{"imageOffset":28584,"symbol":"_pthread_start","symbolLocation":148,"imageIndex":7},{"imageOffset":7584,"symbol":"thread_start","symbolLocation":8,"imageIndex":7}]}],
  "usedImages" : [
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4342317056,
    "CFBundleShortVersionString" : "2.3",
    "CFBundleIdentifier" : "fr.read-write.Submariner",
    "size" : 573440,
    "uuid" : "aac5136e-2ecf-364e-a323-6b623b284a36",
    "path" : "\/Applications\/Submariner.app\/Contents\/MacOS\/Submariner",
    "name" : "Submariner",
    "CFBundleVersion" : "2.3"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 4558405632,
    "size" : 49152,
    "uuid" : "570c31f7-94c6-3b86-ae44-1694db0a4bcd",
    "path" : "\/usr\/lib\/libobjc-trampolines.dylib",
    "name" : "libobjc-trampolines.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 4599267328,
    "CFBundleShortVersionString" : "1.0",
    "CFBundleIdentifier" : "com.apple.AppleMetalOpenGLRenderer",
    "size" : 442368,
    "uuid" : "01596a35-894c-3104-8314-40d3c08f18ca",
    "path" : "\/System\/Library\/Extensions\/AppleMetalOpenGLRenderer.bundle\/Contents\/MacOS\/AppleMetalOpenGLRenderer",
    "name" : "AppleMetalOpenGLRenderer",
    "CFBundleVersion" : "1"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6957740032,
    "CFBundleShortVersionString" : "6.9",
    "CFBundleIdentifier" : "com.apple.CoreFoundation",
    "size" : 5087232,
    "uuid" : "203e4401-8c2e-3157-a24b-92f52551d43e",
    "path" : "\/System\/Library\/Frameworks\/CoreFoundation.framework\/Versions\/A\/CoreFoundation",
    "name" : "CoreFoundation",
    "CFBundleVersion" : "1977"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6974099456,
    "CFBundleShortVersionString" : "6.9",
    "CFBundleIdentifier" : "com.apple.Foundation",
    "size" : 10248192,
    "uuid" : "d5337cc6-a075-3ca0-b5bf-b49da8878337",
    "path" : "\/System\/Library\/Frameworks\/Foundation.framework\/Versions\/C\/Foundation",
    "name" : "Foundation",
    "CFBundleVersion" : "1977"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 7010586624,
    "CFBundleShortVersionString" : "6.9",
    "CFBundleIdentifier" : "com.apple.AppKit",
    "size" : 15790080,
    "uuid" : "041c7d21-21cc-3eea-ae53-d39f5ca6a24e",
    "path" : "\/System\/Library\/Frameworks\/AppKit.framework\/Versions\/C\/AppKit",
    "name" : "AppKit",
    "CFBundleVersion" : "2299.60.124"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6953820160,
    "size" : 582996,
    "uuid" : "bd993f39-c902-37d6-8bd8-6d827c7f121e",
    "path" : "\/usr\/lib\/dyld",
    "name" : "dyld"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6957322240,
    "size" : 53248,
    "uuid" : "46d35233-a051-3f4f-bba4-ba56dddc4d1a",
    "path" : "\/usr\/lib\/system\/libsystem_pthread.dylib",
    "name" : "libsystem_pthread.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6957084672,
    "size" : 237544,
    "uuid" : "42f503e2-9273-360a-a086-c1b19bbd3962",
    "path" : "\/usr\/lib\/system\/libsystem_kernel.dylib",
    "name" : "libsystem_kernel.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 7542820864,
    "CFBundleShortVersionString" : "3.0",
    "CFBundleIdentifier" : "com.apple.imageKit",
    "size" : 2375680,
    "uuid" : "b8b17538-9d8d-304f-9fd0-ef28ab78c686",
    "path" : "\/System\/Library\/Frameworks\/Quartz.framework\/Versions\/A\/Frameworks\/ImageKit.framework\/Versions\/A\/ImageKit",
    "name" : "ImageKit",
    "CFBundleVersion" : "1128.1"
  }
],
  "sharedCache" : {
  "base" : 6953156608,
  "size" : 3551805440,
  "uuid" : "7d9faa84-5c6b-3ef4-9379-faba64346673"
},
  "vmSummary" : "ReadOnly portion of Libraries: Total=1.4G resident=0K(0%) swapped_out_or_unallocated=1.4G(100%)\nWritable regions: Total=1.5G written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=1.5G(100%)\n\n                                VIRTUAL   REGION \nREGION TYPE                        SIZE    COUNT (non-coalesced) \n===========                     =======  ======= \nAccelerate framework               256K        2 \nActivity Tracing                   256K        1 \nCG backing stores                 2176K        4 \nCG image                          8080K       21 \nColorSync                          496K       25 \nCoreAnimation                     24.5M       75 \nCoreData                            48K        1 \nCoreData Object IDs               4112K        2 \nCoreGraphics                        48K        2 \nCoreUI image data                 2832K       23 \nFoundation                          16K        1 \nImage IO                          1024K        1 \nKernel Alloc Once                   32K        1 \nMALLOC                           266.1M       56 \nMALLOC guard page                  288K       15 \nMALLOC_MEDIUM (reserved)         840.0M        7         reserved VM address space (unallocated)\nMALLOC_NANO (reserved)           384.0M        1         reserved VM address space (unallocated)\nSQLite page cache                  128K        1 \nSTACK GUARD                       56.1M        7 \nStack                             11.2M        7 \nVM_ALLOCATE                       1232K       14 \n__AUTH                            1927K      358 \n__AUTH_CONST                      29.2M      582 \n__CTF                               824        1 \n__DATA                            13.0M      568 \n__DATA_CONST                      28.4M      588 \n__DATA_DIRTY                      2066K      226 \n__FONT_DATA                        2352        1 \n__GLSLBUILTINS                    5174K        1 \n__INFO_FILTER                         8        1 \n__LINKEDIT                       802.2M        4 \n__OBJC_RO                         66.3M        1 \n__OBJC_RW                         2011K        1 \n__TEXT                           631.5M      609 \ndyld private memory                272K        2 \nmapped file                      218.8M       39 \nshared memory                      976K       20 \n===========                     =======  ======= \nTOTAL                              3.3G     3269 \nTOTAL, minus reserved VM space     2.1G     3269 \n",
  "legacyInfo" : {
  "threadTriggered" : {
    "queue" : "com.apple.main-thread"
  }
},
  "logWritingSignature" : "e63724b99366d9cdacb8c91519e26201038c973a",
  "trialInfo" : {
  "rollouts" : [
    {
      "rolloutId" : "6297d96be2c9387df974efa4",
      "factorPackIds" : {

      },
      "deploymentId" : 240000008
    },
    {
      "rolloutId" : "60186475825c62000ccf5450",
      "factorPackIds" : {

      },
      "deploymentId" : 240000055
    }
  ],
  "experiments" : [
    {
      "treatmentId" : "c28e4ee6-1b08-4f90-8e05-2809e78310a3",
      "experimentId" : "6317d2003d24842ff850182a",
      "deploymentId" : 400000013
    }
  ]
}
}

Model: MacBookAir10,1, BootROM 10151.0.156.505.1, proc 8:4:4 processors, 16 GB, SMC 
Graphics: Apple M1, Apple M1, Built-In
Display: Color LCD, 2560 x 1600 Retina, Main, MirrorOff, Online
Memory Module: LPDDR4, Hynix
AirPort: spairport_wireless_card_type_wifi (0x14E4, 0x4378), wl0: Jan 12 2023 05:52:26 version 18.20.383.14.7.8.149 FWID 01-1469d19d
Bluetooth: Version (null), 0 services, 0 devices, 0 incoming serial ports
Network Service: Wi-Fi, AirPort, en0
USB Device: USB31Bus
USB Device: USB31Bus
Thunderbolt Bus: MacBook Air, Apple Inc.
Thunderbolt Bus: MacBook Air, Apple Inc.
```

</details>

**Comment by tarasis** [comment]
Plan to grab a copy of the repo so see exactly where / why its crashing when I get a moment.

**Comment by tarasis** [comment]
This is the data for attempting to access the created server in the sidebar. The app also crashes at this point when adding a new server.

```
2023-06-24 17:06:18.565918+0100 Submariner[32254:405665] [SBServer] SBServer.password getter: Getting internet keychain for http://removed user tarasis
2023-06-24 17:06:18.619440+0100 Submariner[32254:405665] [SBServer] SBServer.password getter: Keychain item not found
2023-06-24 17:06:18.619491+0100 Submariner[32254:405665] [SBServer] Base params for http://removed:
2023-06-24 17:06:18.619691+0100 Submariner[32254:405665] [URL+Parameters] Assembling base URL http://removed/rest/ping.view
2023-06-24 17:06:18.619706+0100 Submariner[32254:405665] [URL+Parameters] 	API endpoint /rest/ping.view
2023-06-24 17:06:18.649163+0100 Submariner[32254:405665] [SBServer] SBServer.password getter: Getting internet keychain for http://backup.tarasis.net:4533 user tarasis
2023-06-24 17:06:18.649795+0100 Submariner[32254:405665] [SBServer] SBServer.password getter: Keychain item not found
Submariner/SBClientController.swift:33: Fatal error: Unexpectedly found nil while unwrapping an Optional value
2023-06-24 17:06:18.650031+0100 Submariner[32254:405665] Submariner/SBClientController.swift:33: Fatal error: Unexpectedly found nil while unwrapping an Optional value
(lldb) 
```

Code line

```
let loginString = server.username! + ":" + server.password!
```

(I've removed the url)

----

**Comment by NattyNarwhal** [comment]
I've fixed this in the repo (commit 8b5a8fa0a8ab3d7ba7bc3ef35fb063a8debce4ab), with a build [here](https://github.com/SubmarinerApp/Submariner/files/11858540/Submariner-2.3.zip) you can try.

I think this is a crash just exposed by something else not working though - it looks like it's having issues fetching it from Keychain?

**Comment by tarasis** [comment]
That build did the trick, I can now access the server!

(had to delete the server I had already added, which was tricky because it threw up an error dialog a few times that parameter u was missing)
